src/analysis/retail/evoker/preservation/modules/core/MasteryEffectiveness.tsx

```diff
@@ -266,6 +235,73 @@ class MasteryEffectiveness extends Analyzer {
     this.totalMasteryHealingDone += absorbFromMasteryBonusUsed;
   }
 
+  __masteryBoostCheck(event: ApplyBuffEvent | RefreshBuffEvent, existingAbsorb: number) {
+    const mastery = this.statTracker.currentMasteryPercentage;
+    const baseShieldValue = this.__calculateShield(event, existingAbsorb);
+    const lowerShieldBound = baseShieldValue * (1 + (mastery - 0.1));
+    const upperShieldBound = baseShieldValue * (1 + (mastery + 0.1));
+    if (event.absorb) {
+      if (lowerShieldBound <= event.absorb && event.absorb <= upperShieldBound) {
+        return true;
+      }
+    }
+    return false;
+  }
+  //TODO -- THESE CALCULATIONS ARE USING A +/- 10% THRESHOLD UNTIL ALL THE NEW RAID BUFFS ARE ADDED AND INCORPORATED INTO STATTRACKER
+  __calculateShield(event: ApplyBuffEvent | RefreshBuffEvent, existingAbsorb: number) {
+    let baseShielding = 0;
+    const vers = this.statTracker.currentVersatilityPercentage + 0.03; // TODO: REMOVE WHEN MARK OF THE WILD IS ADDED TO STATTRACKER
+    if (this.__isEventAbilityAnomaly(event.ability.guid)) {
+      const intellect = this.statTracker.currentIntellectRating;
+      baseShielding = intellect * TEMPORAL_ANOMALY_SHIELD_COEFF * (1 + vers) + existingAbsorb;
+    }
+
+    if (this.__isEventAbilityTwinGuardian(event.ability.guid)) {
+      const initialShieldAmount = this.prevokerMaxHealth * TWIN_GUARDIAN_SHIELD_COEFF;
+      const versAddedShield = this.prevokerMaxHealth * TWIN_GUARDIAN_SHIELD_COEFF * vers;
+      baseShielding = initialShieldAmount + versAddedShield + existingAbsorb;
+    }
+    return Math.round(baseShielding);
+  }
+
+  __isTargetHealthierThanPlayer(playerHealth: number, targetHealth: number): boolean {
+    return playerHealth >= targetHealth;
+  }
```

Comment:
this is backwards, and could just be inlined since it is short and clear on its own. (meaning: remove this method, and just do `playerHealth < targetHealth` in the method

Reply:
> okay, so bad news and good news.
> 
> bad news: Mastery Effectiveness as-is doesn't work. You're never going to find existing absorbs, so the thing its trying to do with that isn't actually going to happen in practice.
> 
> good news: there isn't _that_ much to do to make it work.
> 
> this module on the whole could use some comments explaining what the different things are, and what the calculation you're doing is for.

I think I'm going to strip away most of this pre-existing stuff then. The only thing that needs to be determined is whether or not an event had mastery applied to it and what percentage of total events were affected for this module to be valuable. The only hiccup there being refresh events on the absorbs that stack (see other comment below)